### PR TITLE
Add troubleshooting steps for transport endpoint not connected.

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
Added troubleshooting steps for transport endpoint not connected.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - NA

### CODE REVIEW

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -116,3 +117,9 @@ In case a new file is added to the empty directory remotely, outside of the moun
 ### fuse: *fuseops.FallocateOp error: function not implemented or similar function not implemented errors
 
 This is an expected error for file operations unsupported in FUSE file system (details [here](https://github.com/GoogleCloudPlatform/gcsfuse/discussions/2386#discussioncomment-10417635)).
+
+### transport endpoint is not connected
+
```
Comment (charith87) [inline docs/troubleshooting.md:122]: It might be good to what the error is and the reason before adding the solution.

Comment (charith87) [reply docs/troubleshooting.md:122]: cl/642277208 seems to have most of the steps. Could we add similar here. We could also add above the less resources as an additional point.

File: docs/troubleshooting.md [hunk 2]
```diff
@@ -116,3 +117,9 @@ In case a new file is added to the empty directory remotely, outside of the moun
 ### fuse: *fuseops.FallocateOp error: function not implemented or similar function not implemented errors
 
 This is an expected error for file operations unsupported in FUSE file system (details [here](https://github.com/GoogleCloudPlatform/gcsfuse/discussions/2386#discussioncomment-10417635)).
+
+### transport endpoint is not connected
```
Comment (gargnitingoogle) [inline docs/troubleshooting.md:121]:
```suggestion
### Error "transport endpoint is not connected"
```

File: docs/troubleshooting.md (hunk 1 above)
Comment (gargnitingoogle) [reply docs/troubleshooting.md:122]: +1. this will be good for consistency also.

Comment (Tulsishah) [reply docs/troubleshooting.md:124]: Added in the form of cl/642277208

File: docs/troubleshooting.md [hunk 3]
```diff
@@ -116,3 +117,15 @@ In case a new file is added to the empty directory remotely, outside of the moun
 ### fuse: *fuseops.FallocateOp error: function not implemented or similar function not implemented errors
 
 This is an expected error for file operations unsupported in FUSE file system (details [here](https://github.com/GoogleCloudPlatform/gcsfuse/discussions/2386#discussioncomment-10417635)).
+
+### Error "transport endpoint is not connected".
```
Comment (gargnitingoogle) [inline docs/troubleshooting.md:121]: nit: Please remove the trailing `.` This is not a setence, but a label.

Comment (gargnitingoogle) [reply docs/troubleshooting.md:121]:
```suggestion
### Error "transport endpoint is not connected"
```

File: docs/troubleshooting.md [hunk 4]
```diff
@@ -116,3 +117,15 @@ In case a new file is added to the empty directory remotely, outside of the moun
...
+
+It is possible customer is seeing the error "transport endpoint is not connected" because if the previous mount get crashed or a mounted filesystem or device was abruptly disconnected or the mounting process fails unexpectedly, the system might not properly update its mount table. This leaves a "stale" entry referencing a resource that's no longer available.
```
Comment (gargnitingoogle) [inline docs/troubleshooting.md:123]: nit: grammar

s/get crashed/crashed/g
s/fails/failed/g

File: docs/troubleshooting.md [hunk 5]
```diff
@@ -116,3 +117,15 @@ In case a new file is added to the empty directory remotely, outside of the moun
...
+
+**Solution:** Run the command `mount | grep "gcsfuse"`. If you find any entries, unmount the corresponding directory multiple times until all the entries got clear and then try to remount the bucket.
```
Comment (gargnitingoogle) [inline docs/troubleshooting.md:125]:
```suggestion
**Solution:** Run the command `mount | grep "gcsfuse"`. If you find any stale entries, unmount the corresponding directory multiple times until all the entries got clear and then try to remount the bucket.
```

Comment (gargnitingoogle) [inline docs/troubleshooting.md:125]: nit: grammar

s/got clear/cleared up/g

File: docs/troubleshooting.md [hunk 6]
```diff
@@ -116,3 +117,15 @@ In case a new file is added to the empty directory remotely, outside of the moun
...
+
+**Additional troubleshooting steps:**
+
+- Try to unmount and mount the mount-point using the command:
+   `umount -f /<mount point>` & `mount /<mount point>`
```
Comment (gargnitingoogle) [inline docs/troubleshooting.md:130]: This is a bit confusing. Shouldn't this be:

```suggestion
   `umount -f /<mount point> && mount /<mount point>`
```

**Review by charith87 — DISMISSED** [review]

**Review by charith87 — DISMISSED** [review]

**Review by gargnitingoogle — APPROVED** [review]
LGTM

**Review by charith87 — APPROVED** [review]

**Pull request merged**